### PR TITLE
Update AKS recommendations

### PR DESCRIPTION
**Pull request by ibersanoMS**
# Overview/Summary

- Added query for AKS recommendations: 26ebaf1f-c70d-4ebd-8641-4b60a0ce0094, a7bfcc18-b0d8-4d37-81f3-8131ed8bead5 (Using Ephemeral Disks and Enabling Azure Policy)
- Added AKS recommendation and query: ba3f7d17-9654-4038-b172-09f1b2569925 (Private cluster or authorized IP ranges for cluster network resiliency)

## This PR fixes/adds/changes/removes

1. Adds query for AKS recommendations: 26ebaf1f-c70d-4ebd-8641-4b60a0ce0094, a7bfcc18-b0d8-4d37-81f3-8131ed8bead5 (Using Ephemeral Disks and Enabling Azure Policy)
2. Adds AKS recommendation and query: ba3f7d17-9654-4038-b172-09f1b2569925 (Private cluster or authorized IP ranges for cluster network resiliency)

## As part of this pull request I have

- [x] Read the [Contribution Guide](https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing) and ensured this PR is compliant with the guide
- [x] Checked for duplicate [Pull Requests](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/pulls)
- [x] Associated it with relevant [GitHub Issues](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/issues) or ADO Work Items (Internal Only)
- [x] Ensured my code/branch is up-to-date with the latest changes in the `main` [branch](https://github.com/Azure/Azure-Proactive-Resiliency-Library-v2/tree/main)
- [x] Ensured PR tests are passing
- [x] Performed testing and provided evidence (e.g. screenshot of output) for any changes associated to ARG queries
- [x] Updated relevant and associated documentation (e.g. Contribution Guide, Docs etc.)

### CODE REVIEW

**Comment by Dost2010** [comment]
Hello @ibersanoMS 
Thank you for your contribution. The long description for recommendation "Enable AKS private cluster or set authorized IP address ranges" is more than 300 characters. Can you please rewrite the long description, so it does fit in less than 300 characters. 
For more information, please refer to contribution guide https://azure.github.io/Azure-Proactive-Resiliency-Library-v2/contributing/create-content/create-recommendations/

**Comment by Dost2010** [comment]
The long description is still 313 characters.

**Comment by ibersanoMS** [comment]
It's 215 - there was an indentation mistake so potentialBenefits got pulled up to the previous line in YAML. It's corrected now.

File: azure-resources/ContainerService/managedClusters/recommendations.yaml [hunk 1]
```diff
@@ -501,3 +501,24 @@
...
+- description: Enable AKS private cluster or set authorized IP address ranges
+  aprlGuid: ba3f7d17-9654-4038-b172-09f1b2569925
+  recommendationTypeId: null
+  recommendationControl: High Availability
+  recommendationImpact: High
+  recommendationResourceType: Microsoft.ContainerService/managedClusters
+  recommendationMetadataState: Active
+  longDescription: |
+    For optimal connectivity and security, enable private cluster in AKS to ensure direct communication between worker nodes and the control plane. If not enabled, secure API server access with authorized IP ranges.
+  potentialBenefits: Provides network security and reliability, ensures stability, reduced disruptions
```
Comment (ejhenry) [inline azure-resources/ContainerService/managedClusters/recommendations.yaml:514]: Please reword potentialBenefits to bring under 60 characters, including spaces. Current count is 81 characters

File: azure-resources/ContainerService/managedClusters/recommendations.yaml [hunk 2]
```diff
@@ -501,3 +501,24 @@
...
+  pgVerified: false
+  publishedToLearn: false
+  publishedToAdvisor: false
+  automationAvailable: arg
+  tags: null
+  learnMoreLink:
```
Comment (ejhenry) [inline azure-resources/ContainerService/managedClusters/recommendations.yaml:520]: Please include only a single learn more link

File: azure-resources/ContainerService/managedClusters/kql/26ebaf1f-c70d-4ebd-8641-4b60a0ce0094.kql
Comment (ejhenry) [inline azure-resources/ContainerService/managedClusters/kql/26ebaf1f-c70d-4ebd-8641-4b60a0ce0094.kql:1]: Please ensure query output contains required columns

File: azure-resources/ContainerService/managedClusters/kql/a7bfcc18-b0d8-4d37-81f3-8131ed8bead5.kql [hunk 3]
```diff
@@ -1,2 +1,8 @@
-// cannot-be-validated-with-arg
-
+// Azure Resource Graph Query
+// Returns any AKS cluster nodepools that do not have Ephemeral Disks
+resources
+| where type == "microsoft.containerservice/managedclusters"
+| mv-expand agentPoolProfile = properties.agentPoolProfiles
+| extend type = tostring(agentPoolProfile.osDiskType)
+| where type != 'Ephemeral'
+| project recommendationid="a7bfcc18-b0d8-4d37-81f3-8131ed8bead5", id, name, param1=strcat("osDiskType: ", type)
```
Comment (ejhenry) [inline azure-resources/ContainerService/managedClusters/kql/a7bfcc18-b0d8-4d37-81f3-8131ed8bead5.kql:8]:
```suggestion
| project recommendationid="a7bfcc18-b0d8-4d37-81f3-8131ed8bead5", name, id, param1=strcat("osDiskType: ", type)
```

File: azure-resources/ContainerService/managedClusters/kql/ba3f7d17-9654-4038-b172-09f1b2569925.kql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// Azure Resource Graph Query
+// Returns clusters that do not have private cluster or authorized IP ranges set up (both are not set)
+resources
+| where type == 'microsoft.containerservice/managedclusters'
+| mv-expand properties.apiServerAccessProfile
+| where isempty(properties.apiServerAccessProfile.enablePrivateCluster) and isempty(properties.apiServerAccessProfile.authorizedIPRanges)
+| project recommendationId="ba3f7d17-9654-4038-b172-09f1b2569925", id, name
```
Comment (ejhenry) [inline azure-resources/ContainerService/managedClusters/kql/ba3f7d17-9654-4038-b172-09f1b2569925.kql:7]:
```suggestion
| project recommendationId="ba3f7d17-9654-4038-b172-09f1b2569925", name, id
```

**Review by ejhenry — COMMENTED** [review]
Thanks for the contribution, @ibersanoMS! Just a couple items that need to be addressed to align with the APRLv2 structure.

**Review by ejhenry — APPROVED** [review]

**Review by Dost2010 — APPROVED** [review]

**Pull request merged**